src/Nethermind/Ethereum.Test.Base/BlockchainTestBase.cs

```diff
@@ -289,6 +237,56 @@ protected async Task<EthereumTestResult> RunTest(BlockchainTest test, Stopwatch?
             );
         }
 
+        private List<(Block Block, string ExpectedException)> DecodeRlps(BlockchainTest test)
+        {
+            List<(Block Block, string ExpectedException)> correctRlp = new();
+            for (int i = 0; i < test.Blocks.Length; i++)
+            {
+                TestBlockJson testBlockJson = test.Blocks[i];
+                try
+                {
+                    var rlpContext = Bytes.FromHexString(testBlockJson.Rlp).AsRlpStream();
+                    Block suggestedBlock = Rlp.Decode<Block>(rlpContext);
+                    suggestedBlock.Header.SealEngineType =
+                        test.SealEngineUsed ? SealEngineType.Ethash : SealEngineType.None;
+
+                    if (testBlockJson.BlockHeader is not null)
+                    {
+                        Assert.AreEqual(new Keccak(testBlockJson.BlockHeader.Hash), suggestedBlock.Header.Hash,
+                            "hash of the block");
+
+
+                        for (int uncleIndex = 0; uncleIndex < suggestedBlock.Uncles.Length; uncleIndex++)
+                        {
+                            Assert.AreEqual(new Keccak(testBlockJson.UncleHeaders[uncleIndex].Hash),
+                                suggestedBlock.Uncles[uncleIndex].Hash, "hash of the uncle");
+                        }
+
+                        correctRlp.Add((suggestedBlock, testBlockJson.ExpectedException));
+                    }
+                }
+                catch (Exception e)
+                {
+                    if (testBlockJson.ExpectedException is null)
+                    {
+                        Assert.Fail($"Invalid RLP ({i}) {e}");
+                    }
+                    else
+                    {
+                        _logger.Info($"Expected invalid RLP ({i})");
```

Comment:
_logger.Error?

Reply:
Should test fail in this case or not?

Reply:
No, this block have to be rejected, so it's just info